mvlearn/decomposition/multiviewica.py

```diff
@@ -0,0 +1,344 @@
+            tol=self.tol,
+            verbose=self.verbose,
+        )
+        mixing_ = np.array([np.linalg.pinv(W) for W in unmixings_])
+        self.components_ = unmixings_
+        self.mixing_ = mixing_
+        if preprocessing_name == "pca":
+            pca_components = []
+            for i, transformer in enumerate(
+                self.preproc_instance_.transformers_
+            ):
+                K = transformer.components_
+                pca_components.append(K)
+            self.pca_components_ = np.array(pca_components)
+
+        if preprocessing_name == "None":
+            self.individual_components_ = unmixings_
+            self.individual_mixing_ = mixing_
+        else:
+            self.individual_mixing_ = []
+            self.individual_components_ = []
+            sources_pinv = linalg.pinv(S.T)
+            for x in Xs:
+                lstq_solution = np.dot(sources_pinv, x)
+                self.individual_components_.append(
+                    linalg.pinv(lstq_solution).T
+                )
+                self.individual_mixing_.append(lstq_solution.T)
+        return self
+
+    def transform(self, X):
+        r"""
+        Recover the sources from each view (apply unmixing matrix).
+
+        Parameters
+        ----------
+        Xs : list of array-likes or numpy.ndarray
+            - Xs length: n_views
+            - Xs[i] shape: (n_samples, n_features_i)
+            Training data to recover a source and unmixing matrices from.
+
+        Returns
+        -------
+        Xs_new : numpy.ndarray, shape (n_views, n_samples, n_components)
+            The mixed sources from the single source and per-view unmixings.
+        """
+        if not hasattr(self, "components_"):
+            raise ValueError("The model has not yet been fitted.")
+
+        if self.preproc_instance_ is not None:
+            transformed_X = self.preproc_instance_.transform(X)
+        else:
+            transformed_X = X.copy()
+        if self.multiview_output:
+            return np.array(
+                [x.dot(w.T) for w, x in zip(self.components_, transformed_X)]
+            )
+        else:
+            return np.mean(
+                [x.dot(w.T) for w, x in zip(self.components_, transformed_X)],
+                axis=0,
+            )
+
+    def inverse_transform(self, X_transformed):
+        r"""
+        Transforms the sources back to the mixed data for each view
+        (apply mixing matrix).
+
+        Parameters
+        ----------
+        X_transformed : list of array-likes or numpy.ndarray
+            The dataset corresponding to transformed data
+
+
+
+
+
+        Returns
+        -------
+        Xs_new : numpy.ndarray, shape (n_views, n_samples, n_components)
+            The mixed sources from the single source and per-view unmixings.
+        """
+        if not hasattr(self, "components_"):
+            raise ValueError("The model has not yet been fitted.")
```

Comment:
you could use the check_is_fitted function:

https://scikit-learn.org/stable/modules/generated/sklearn.utils.validation.check_is_fitted.html